test/integration/scheduler/preemption_test.go

```diff
@@ -434,6 +437,120 @@ func TestPreemption(t *testing.T) {
+					Resources: &v1.ResourceRequirements{Requests: v1.ResourceList{
+						v1.ResourceCPU:    *resource.NewMilliQuantity(400, resource.DecimalSI),
+						v1.ResourceMemory: *resource.NewQuantity(200, resource.DecimalSI)},
+					},
+				}),
+			},
+			pod: initPausePod(cs, &pausePodConfig{
+				Name:             "preemptor-pod",
+				Namespace:        testCtx.NS.Name,
+				Priority:         &highPriority,
+				PreemptionPolicy: &preemptNever,
+				Resources: &v1.ResourceRequirements{Requests: v1.ResourceList{
+					v1.ResourceCPU:    *resource.NewMilliQuantity(300, resource.DecimalSI),
+					v1.ResourceMemory: *resource.NewQuantity(200, resource.DecimalSI)},
+				},
+			}),
+		},
+		{
+			description: "pod preemption will happen",
+			happen:      true,
+			existingPods: []*v1.Pod{
+				initPausePod(testCtx.ClientSet, &pausePodConfig{
+					Name:      "victim-pod",
+					Namespace: testCtx.NS.Name,
+					Priority:  &lowPriority,
+					Resources: &v1.ResourceRequirements{Requests: v1.ResourceList{
+						v1.ResourceCPU:    *resource.NewMilliQuantity(400, resource.DecimalSI),
+						v1.ResourceMemory: *resource.NewQuantity(200, resource.DecimalSI)},
+					},
+				}),
+			},
+			pod: initPausePod(cs, &pausePodConfig{
+				Name:      "preemptor-pod",
+				Namespace: testCtx.NS.Name,
+				Priority:  &highPriority,
+				Resources: &v1.ResourceRequirements{Requests: v1.ResourceList{
+					v1.ResourceCPU:    *resource.NewMilliQuantity(300, resource.DecimalSI),
+					v1.ResourceMemory: *resource.NewQuantity(200, resource.DecimalSI)},
+				},
+			}),
+		},
+	}
+
+	// Create a node with some resources and a label.
+	nodeRes := &v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewMilliQuantity(500, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(500, resource.DecimalSI),
+	}
+	_, err := createNode(testCtx.ClientSet, "node1", nodeRes)
+	if err != nil {
+		t.Fatalf("Error creating nodes: %v", err)
+	}
+
+	for _, test := range tests {
+		pods := make([]*v1.Pod, len(test.existingPods))
+		// Create and run existingPods.
+		for i, p := range test.existingPods {
+			pods[i], err = runPausePod(cs, p)
+			if err != nil {
+				t.Fatalf("Test [%v]: Error running pause pod: %v", test.description, err)
+			}
+		}
+		// Create the "pod".
+		preemptor, err := createPausePod(cs, test.pod)
+		if err != nil {
+			t.Errorf("Error while creating high priority pod: %v", err)
+		}
+		// Ensure preemptor should keep unschedulable.
+		if err := waitForPodUnschedulable(cs, preemptor); err != nil {
+			t.Errorf("Test [%v]: Preemptor %v should not become scheduled",
+				test.description, preemptor.Name)
+		}
+
+		// Check if preemptor is nominated.
+		err = waitForNominatedNodeNameWithTimeout(cs, preemptor, 5*time.Second)
+		if err != nil && test.happen {
+			t.Errorf("Test [%v]: Preemptor %v should be nominated",
+				test.description, preemptor.Name)
+		} else if err == nil && !test.happen {
```

Comment:
In this case, the test will finish after 30 seconds, right? As you're keep waiting until the default timeout (30 seconds).

If so, I'd recommend to change the case like this:

- Set 0 terminationSecond to each Pod, so that the victim can be deleted physically
- Create 3 Pods with different priority (high, med, low), with each request for 200 CPU. Create low and med Pods first to ensure they're scheduled. Finally create high Pod.
    - case1: low pod with &preemptNever - verify if med pod gets preempted, and then high pod gets scheduled.
    - case2: low pod without &preemptNever - verify if low pods gets preempted, and then high pod gets scheduled.

Reply:
The test will finish less than 2*5s.

Reply:
> Create 3 Pods with different priority (high, med, low), with each request for 200 CPU. Create low and med Pods first to ensure they're scheduled. Finally create high Pod.
case1: low pod with &preemptNever - verify if med pod gets preempted, and then high pod gets scheduled.
case2: low pod without &preemptNever - verify if low pods gets preempted, and then high pod gets scheduled.

There may be some misunderstanding here. the NonPreempt indicates whether the pod can preempt others. Not whether the pod can be preempted by others. @Huang-Wei